tensorflow/c/c_api.h

```diff
@@ -529,16 +529,23 @@ typedef struct TF_AttrMetadata {
 // Returns metadata about the value of the attribute `attr_name` of `oper`.
 TF_CAPI_EXPORT extern TF_AttrMetadata TF_OperationGetAttrMetadata(
     TF_Operation* oper, const char* attr_name, TF_Status* status);
+TF_CAPI_EXPORT extern TF_AttrMetadata TF_OperationGetAttributeMetadata(
+    TF_Operation* oper, const char* attr_name, TF_Status* status);
 
 // Fills in `value` with the value of the attribute `attr_name`.  `value` must
 // point to an array of length at least `max_length` (ideally set to
-// TF_AttrMetadata.total_size from TF_OperationGetAttrMetadata(oper,
+// TF_AttrMetadata.total_size from TF_OperationGetAttributeMetadata(oper,
 // attr_name)).
 TF_CAPI_EXPORT extern void TF_OperationGetAttrString(TF_Operation* oper,
                                                      const char* attr_name,
                                                      void* value,
                                                      size_t max_length,
                                                      TF_Status* status);
+TF_CAPI_EXPORT extern void TF_OperationGetAttributeString(TF_Operation* oper,
```

Comment:
I think users just reading this header file would be very confused by seemingly similar functions, and wouldn't know what's the difference / what they are encouraged to use.

Could you move all the depreciated functions to `c_api_deprecated.h` and import from this header?  Also, please let's comment why it's deprecated and also have a link to this PR.